packages/amplify-provider-awscloudformation/src/graphql-transformer/transform-graphql-schema.ts

```diff
@@ -563,3 +567,36 @@ async function _buildProject(opts: ProjectOptions<TransformerFactoryArgs>) {
 
   return mergeUserConfigWithTransformOutput(userProjectConfig, transformOutput, opts);
 }
+
+async function notifySecurityEnhancement(context) {
+  if (FeatureFlags.getBoolean('graphqltransformer.securityEnhancementNotification')) {
+    const meta = stateManager.getMeta();
+
+    const apiNames = Object.entries(stateManager.getMeta()?.api || {})
+      .filter(([_, apiResource]) => (apiResource as $TSAny).service === 'AppSync')
+      .map(([name]) => name);
+
+    if (meta.auth && apiNames.length === 1) {
+      const projectPath = pathManager.findProjectRoot() ?? process.cwd();
+
+      let config = stateManager.getCLIJSON(projectPath, undefined, {
+        throwIfNotExist: false,
+        preserveComments: true,
+      });
+
+      const apiName = apiNames[0];
+      const apiResourceDir = path.join(pathManager.getBackendDirPath(), 'api', apiName);
+      const schemaPath = path.join(apiResourceDir, 'schema.graphql');
+
+      fs.appendFile(schemaPath, ' ');
+
+      config.features.graphqltransformer.securityEnhancementNotification = false;
+      stateManager.setCLIJSON(projectPath, config);
+      await FeatureFlags.reloadValues();
+
+      context.print.info(
+        `\nThis version of Amplify CLI introduces additional security enhancements for your GraphQL API. @auth authorization rules applied on primary keys and indexes are scoped down further. The changes are applied automatically with this deployment. This change won't impact your client code.`,
```

Comment:
nit: use printer.blankLine() instead of starting the string with `\n`

Reply:
updated